apps/docs/components/ProjectConfigVariables.tsx

```diff
@@ -0,0 +1,455 @@
+
+  if (organizationName && projectName) {
+    return (
+      instances.find(
+        (instance) =>
+          // This gets filtered through `Command`'s `value` prop,
+          // which automatically converts to lowercase
+          instance.organization?.name.toLowerCase() === organizationName &&
+          instance.project?.name.toLowerCase() === projectName &&
+          (!instance.branch || instance.branch.name.toLowerCase() === branchName)
+      ) ?? null
+    )
+  }
+
+  return null
+}
+
+const instanceStore = proxy({
+  selectedInstanceKey: null as InstanceKey | null,
+  setSelectedInstanceKey: (instanceKey: InstanceKey | null) => {
+    instanceStore.selectedInstanceKey = instanceKey
+    if (instanceKey !== null) {
+      store('local', LOCAL_STORAGE_KEYS.SAVED_ORG_PROJECT_BRANCH, toStoredString(instanceKey))
+    }
+  },
+  clear: () => {
+    instanceStore.setSelectedInstanceKey(null)
+  },
+})
+
+type Variable = 'url' | 'anonKey'
+
+const prettyFormatVariable: Record<Variable, string> = {
+  url: 'Project URL',
+  anonKey: 'Anon key',
+}
+
+type ProjectConfigVariablesState =
+  | 'userLoading'
+  | 'loggedOut'
+  | 'loggedIn.dataPending'
+  | 'loggedIn.hasData'
+  | 'loggedIn.hasNoData'
+  | 'loggedIn.dataError'
+
+type ProjectConfigVariablesPlusApiState =
+  | Omit<ProjectConfigVariablesState, 'loggedIn.hasData'>
+  | 'loggedIn.hasData.apiDataPending'
+  | 'loggedIn.hasData.apiDataSuccess'
+  | 'loggedIn.hasData.apiDataError'
+
+function ComboBox({
+  parentStateSummary,
+  instances,
+  className,
+}: {
+  parentStateSummary: ProjectConfigVariablesState
+  instances: InstanceData[]
+  className?: string
+}) {
+  const { selectedInstanceKey, setSelectedInstanceKey } = useSnapshot(instanceStore)
+  const selectedInstance = getInstanceDataFromKey(instances, selectedInstanceKey)
+
+  const [open, setOpen] = useState(false)
+
+  return (
+    <Popover open={open} onOpenChange={setOpen}>
+      <div className={cn('flex items-center', className)}>
+        <PopoverTrigger asChild aria-label={`Select project`}>
+          <Button
+            variant="outline"
+            role="combobox"
+            aria-expanded={open}
+            className="overflow-hidden h-auto min-h-10 flex justify-between border-none px-1 text-left"
+          >
+            {parentStateSummary === 'userLoading' || parentStateSummary === 'loggedIn.dataPending'
+              ? 'Loading...'
+              : toPrettyString(selectedInstance) || 'Select a project...'}
+            <ChevronsUpDown className="ml-2 h-4 w-4 shrink-0 opacity-50" />
+          </Button>
+        </PopoverTrigger>
+      </div>
+      <PopoverContent className="z-[999999] p-0" side="bottom">
+        <Command>
+          <CommandInput placeholder="Search project..." className="border-none ring-0" />
+          <CommandEmpty>No project found.</CommandEmpty>
+          <CommandGroup>
+            {instances.map((instance) => (
```

Comment:
We probably need to add a scrollarea here
![image](https://github.com/supabase/supabase/assets/19742402/23351a80-deeb-4fbd-9b80-a42d9d536d23)

can refer to https://github.com/supabase/supabase/blob/1fb755bfc715446de9600ce84084f1b3290b2829/apps/studio/components/layouts/AppLayout/ProjectDropdown.tsx#L138